### PR TITLE
Upgrade Clap to version 3 and bump MSRV to 1.54

### PR DESCRIPTION
**Pull request by mati865**

### CODE REVIEW

**Comment by mati865** [comment]
Unfortunately this requires bumping MSRV to 1.52.

**Comment by mati865** [comment]
I'll corect commit mesage and PR title when I'm back home.

File: README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ extern "C" {
 
 ## MSRV
 
-The minimum supported Rust version is **1.46**.
+The minimum supported Rust version is **1.52**.
```
Comment (slyshykO) [inline README.md:43]: I think there should be `1.54`

Comment (mati865) [reply README.md:43]: Good catch, squashed the change.

**Review by emilio — APPROVED** [review]
Loos fair, thank you!

**Pull request merged**